### PR TITLE
Update ufs-weather-model to 2024-12-06 commit 

### LINKED ISSUE #3110: Issue building pre/post wave jobs with latest ufs-weather-model
### What is wrong?

FYI - There is an issue (seen here: https://github.com/NOAA-EMC/global-workflow/pull/3106#issuecomment-2484169044) when building the wave model if you update to the latest ufs-weather-model.  Fix for this is being tracked here: https://github.com/ufs-community/ufs-weather-model/issues/2502 

Just posting here so others know about this problem. 

### What should have happened?

WW3 pre/post jobs should build. 

### What machines are impacted?

All or N/A

### What global-workflow hash are you using?

adding newest ufs-weather-model 

### Steps to reproduce

Update ufs-weather-model to latest PR and try to build ww3 pre/post. 

### Additional information

_No response_

### Do you have a proposed solution?

_No response_

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
# Description

This PR updates ufs-weather-model hash to the latest commit which addresses an issue seen in an earlier commit (see: https://github.com/NOAA-EMC/global-workflow/issues/3110) 

Note this includes the PR that enables PIO for WW3, however additional work is required to use this feature. 
Another notable PR update is for a PIO finalize bug in CICE which also required updates to ice_in 

Fixes #3110

# Type of change
- [ ] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [x] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO (If YES, please add a link to any PRs that are pending.)
  - [ ] EMC verif-global 
  - [ ] GDAS 
  - [ ] GFS-utils 
  - [ ] GSI 
  - [ ] GSI-monitor 
  - [ ] GSI-utils 
  - [ ] UFS-utils 
  - [x] UFS-weather-model 
  - [ ] wxflow 

# How has this been tested?
Quick test that forecast runs for S2SW GFS.  Running a quick cycled test but wanted to share this now so others don't have to make the same updates here in their local branches. 

# Checklist
- [x] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have documented my code, including function, input, and output descriptions
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

### CODE REVIEW

**Review by aerorahul — DISMISSED** [review]
looks straight-forward.

**Review by DavidHuber-NOAA — DISMISSED** [review]

**Comment by JessicaMeixner-NOAA** [comment]
Sorry I just was checking some tests I ran over the weekend for the WCDA - and I found an issue so we might want to stop the CI as I'm assuming itll find the same issue! 

For the WCDA test: 

The log file is here: 
/scratch1/NCEPDEV/climate/Jessica.Meixner/gwprupdateufs/twcda01/COMROOT/twcda01/logs/2021032412/gdas_fcst_seg0.log.0 and we get an error: 

```
20:
20:  (abort_ice)ABORTED:
20:  (abort_ice) error =
20:  (construct_filename) ERROR: history filename already used for another history s
20:  tream iceh_inst.2021-03-24-43200.nc
```


The ice_in file is here: /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/twcda01/gdas.2021032412/gdasfcst.2021032412/fcst.548096/ice_in

There is one difference between GDAS and GFS runs: 
https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/parsing_namelists_CICE.sh#L74-L78 

So I don't know if maybe the new CICE_hist_suffix should also have a GFS/GDAS setting?   @DeniseWorthen @NickSzapiro-NOAA any chance you might have insight to this?

**Comment by DavidHuber-NOAA** [comment]
@JessicaMeixner-NOAA Okie dokie, I killed the CI on Hercules.

**Comment by NickSzapiro-NOAA** [comment]
@JessicaMeixner-NOAA Yes, with
```
    histfreq            = 'm','d','h','1','x'
    histfreq_n          = 0, 0, 3, 1, 1
    hist_avg            = .false., .false., .false., .false., .false.
```
there are two histfreq/streams trying to write to instantaneous output to the same filename for 2021-03-24-43200

If you don't need output every timestep, you can turn that off with `histfreq_n = 0, 0, 3, 0, 1`

If you do, then you can create different files using the exposed hist_suffix setting to add hist_suffix to the iceh_inst filename. You can choose what you want (e.g., `hist_suffix = 'x','x','h','x','x'` ) where 'x' doesn't add anything

**Comment by JessicaMeixner-NOAA** [comment]
Thanks @NickSzapiro-NOAA I see that in https://github.com/ufs-community/ufs-weather-model/pull/2437 
histfreq            = 'm','d','h','x','x'  
was changed to 
histfreq            = 'm','d','h','1','x'

So we need to add 0s in our      histfreq_n  as you suggest  - I'll push that change now.

**Comment by NickSzapiro-NOAA** [comment]
@JessicaMeixner-NOAA I'm not sure what files folks want but something like:
```
if [[ "${RUN}" =~ "gdas" ]]; then
  local CICE_HIST_AVG=".false., .false., .false., .false., .false."   # DA needs instantaneous
  local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, 0, 0"
else
  local CICE_HIST_AVG=".true., .true., .true., .true., .true."    # GFS long forecaset wants averaged over CICE_HISTFREQ_N
  local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, ${CICE_DIAGFREQ}, 0"
fi
```
would keep the daily-averaged CICE output that g-w wanted too.

**Comment by JessicaMeixner-NOAA** [comment]
@NickSzapiro-NOAA I don't think the GFS job is looking for time step at every output time either, so I'm just changing: 

 local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, 1, 0" 
to 
local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, 0, 0" 
for all cases.  Not sure why it worked for GFS and not GEFS... but maybe that's because of the averages?

**Comment by NickSzapiro-NOAA** [comment]
@JessicaMeixner-NOAA If hist_avg is .true. then the filenames have unique "suffix" already and there are no conflicts

I updated my comment above but the CICE timestep output (https://github.com/ufs-community/ufs-weather-model/pull/2437) was put in for g-w (https://github.com/NOAA-EMC/global-workflow/issues/2674)

**Comment by JessicaMeixner-NOAA** [comment]
Thanks @NickSzapiro-NOAA  


Pinging @bingfu-NOAA and @EricSinsky-NOAA  on this change - since this is related to an open issue related to GEFS.

**Comment by EricSinsky-NOAA** [comment]
@JessicaMeixner-NOAA The changes you made to `ush/parsing_namelists_CICE.sh` look good for cases where `FHOUT_ICE` is `6`. The current configuration in `ush/parsing_namelists_CICE.sh` is problematic when `cycle` is not `00z` and `FHOUT_ICE` is `24`, which is what issue #2674 is addressing. This issue should not affect the CI testing for GEFS since `FHOUT_ICE` is `6` by default. If a user wants to run a case initialized at 12z and output 24-hr averaged CICE history data (averaged from 12z to 12z), I believe the following:
```
local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, 0, 1"
```
would need to change to:

```
local CICE_HISTFREQ_N="0, 0, 0, 72, 1"
```

`72` (or [`FHOUT_ICE`*3600]/`DELTIM`) is the number of timesteps in 24 hours, assuming the time step is 20 minutes. The important thing is that a user now has the ability to output 24-hour average (12z-12z) CICE data by simply modifying `parsing_namelists_CICE.sh`, but I think it is ok to not use this as the default configuration for now (at least for GEFS). I believe this PR will resolve issue #2674.

**Comment by JessicaMeixner-NOAA** [comment]
@EricSinsky-NOAA  thanks for taking a quick look at this. It seems like a separate "if GEFS case" is needed.  If you can provide that code I'm happy to add it here or alternatively it seems like it's acceptable to just wait for a future PR?

**Comment by JessicaMeixner-NOAA** [comment]
@DavidHuber-NOAA  My WCDA test is running and past the earlier failure.  I've also merged in the latest changes now.  I'll let the code managers decide when to start testing CI again - but let me know if theres anything additional from my end that's needed.

**Comment by EricSinsky-NOAA** [comment]
> @EricSinsky-NOAA thanks for taking a quick look at this. It seems like a separate "if GEFS case" is needed. If you can provide that code I'm happy to add it here or alternatively it seems like it's acceptable to just wait for a future PR?

Thank you, @JessicaMeixner-NOAA. I think this special case (`cycle=t12z`, `FHOUT_ICE=24`) can be addressed in a separate PR. I may leave issue #2674 open after further testing of these special cases, which I think can be done after this ufs-weather-model hash update is in g-w develop.

**Comment by DavidHuber-NOAA** [comment]
@JessicaMeixner-NOAA @aerorahul The `nam_micro_lookup.dat` file was moved in the UPP from the `parm/` directory to `fix/`, so `link_workflow.sh` needs to be updated.  See commit https://github.com/NOAA-EMC/UPP/commit/a6c1a38c046aee1ca61b65b5f09f31f4f729a445.  Should this be moved to `fix/` in the global workflow?

**Comment by JessicaMeixner-NOAA** [comment]
Thanks @DavidHuber-NOAA  - It does indeed look like we need some updates.  @WenMeng-NOAA  can you confirm these changes needed for UPP?  I'll start making an update now.

**Comment by JessicaMeixner-NOAA** [comment]
I've updated the link script and the gdas_atmanlupp job succeeded.  My archive failed because I forgot to switch from emc-global to an account I have access to.

**Comment by aerorahul** [comment]
> Thanks @DavidHuber-NOAA - It does indeed look like we need some updates. @WenMeng-NOAA can you confirm these changes needed for UPP? I'll start making an update now.

@JessicaMeixner-NOAA I can confirm these changes are required for the UPP.  They performed some house cleanup and moved files from parm/ to fix/. 
Thanks for including this necessary update

**Comment by JessicaMeixner-NOAA** [comment]
I'm getting shell check warnings - let me know if you would like to see any changes @aerorahul @DavidHuber-NOAA

File: sorc/link_workflow.sh
Comment (aerorahul) [inline sorc/link_workflow.sh]: I think we are already in this directory.  L147

File: sorc/link_workflow.sh
Comment (aerorahul) [inline sorc/link_workflow.sh]: Since this is a UPP fix file, should this be linked in the `HOMEgfs/fix`? @DavidHuber-NOAA

Comment (DavidHuber-NOAA) [reply sorc/link_workflow.sh]: I don't think so.  All of the files in `HOMEgfs/fix` are linked to `glopara/fix` with the exception of those in `fix/product`, which are part of the global workflow and not linked in, so I'm not sure if that follows convention.  The only other case where we link in fix files from a submodule is the gsi-monitor to `HOMEgfs/parm/monitor`, so I think keeping it in parm is fine.

**Comment by WalterKolczynski-NOAA** [comment]
> I'm getting shell check warnings - let me know if you would like to see any changes @aerorahul @DavidHuber-NOAA

You should resolve the shellcheck warnings. Looks like you just need to get rid of the one-element loops.

**Comment by WalterKolczynski-NOAA** [comment]
NB: ufs-community/ufs-weather-model/issues/2502 wasn't closed when the fix was merged into UFS because the original PR got merged into another PR that was not updated to reflect the issue resolution.

**Review by WalterKolczynski-NOAA — DISMISSED** [review]

**Review by DavidHuber-NOAA — DISMISSED** [review]

**Comment by WenMeng-NOAA** [comment]
> Thanks @DavidHuber-NOAA - It does indeed look like we need some updates. @WenMeng-NOAA can you confirm these changes needed for UPP? I'll start making an update now.

@JessicaMeixner-NOAA The UPP related updates in link_workflow.sh look good to me. Thanks!

**Comment by TerrenceMcGuinness-NOAA** [comment]
```
Gather Rocoto statistics
(C96_S2SWA_gefs_replay_ics_32ec7b29 on Hercules)
	Total Cycles: 1
	Number Cycles done: 1
	State: DONE
Experiment C96_S2SWA_gefs_replay_ics_32ec7b29 Completed 1 Cycles: *SUCCESS* at Tue Dec 10 22:46:56 CST 2024
```

CI Experiment passed for case **C96_S2SWA_gefs_replay_ics** for this PR but the Jenkins failed on remote clean up.

```
/work2/noaa/global/CI/HERCULES/3145/gfs/ci/scripts/utils/ci_utils_wrapper.sh cleanup_experiment /work2/noaa/global/CI/HERCULES/3145/RUNTESTS/EXPDIR/C96_S2SWA_gefs_replay_ics_32ec7b29
Shell Script 2.3 sec Found unhandled java.io.IOException exception: Remote call on Hercules-EMC failed
```

Experiment **C48_ATM_32ec7b29** Completed 2 Cycles: *SUCCESS* at Tue Dec 10 16:29:45 CST 2024
Experiment **C96_atm3DVar_32ec7b29** Completed 3 Cycles: *SUCCESS* at Tue Dec 10 19:00:57 CST 2024
Experiment **C96C48_hybatmDA_32ec7b29** Completed 3 Cycles: *SUCCESS* at Tue Dec 10 19:01:13 CST 2024
Experiment **C96_S2SWA_gefs_replay_ics_32ec7b29** Completed 1 Cycles: *SUCCESS* at Tue Dec 10 22:46:56 CST 2024

The remaining two cases **C48_S2SWA_gefs** and **C48_S2SW** are waiting in the Priority Queue

**Review by aerorahul — DISMISSED** [review]
lgtm

File: sorc/link_workflow.sh
Comment (aerorahul) [reply sorc/link_workflow.sh]: works for me.

**Comment by TerrenceMcGuinness-NOAA** [comment]
@JessicaMeixner-NOAA This CI test is still running nominally ...

Experiment C48_ATM_32ec7b29 Completed 2 Cycles: *SUCCESS* at Tue Dec 10 16:29:45 CST 2024
Experiment C96_atm3DVar_32ec7b29 Completed 3 Cycles: *SUCCESS* at Tue Dec 10 19:00:57 CST 2024
Experiment C96C48_hybatmDA_32ec7b29 Completed 3 Cycles: *SUCCESS* at Tue Dec 10 19:01:13 CST 2024
Experiment C96_S2SWA_gefs_replay_ics_32ec7b29 Completed 1 Cycles: *SUCCESS* at Tue Dec 10 22:46:56 CST 2024

The two remaining cases **C48_S2SW** and **C48_S2SWA_gefs** are languishing in the Priority queues on Hercules

File: ush/parsing_ufs_configure.sh
Comment (DeniseWorthen) [inline ush/parsing_ufs_configure.sh]: The restart_from_binary option is only valid if you're also using restartnc. It allows you start from a binary but only write restarts in netcdf going forward.

If restartnc is false, then the option is not exercised.

Comment (JessicaMeixner-NOAA) [reply ush/parsing_ufs_configure.sh]: The PIO restart and binary restarts are not yet being used.  We are still using the inp file.  @sbanihash is working on moving to nml and then we can turn on this feature.  This was not done for this PR to expedite getting other fixes in.

Comment (DeniseWorthen) [reply ush/parsing_ufs_configure.sh]: I understand that, but you're setting the restart_from_binary=true, which implies you're asking the code to do something. But this setting is not pertinent w/o using restartnc. I would leave it false.

**Comment by WalterKolczynski-NOAA** [comment]
Holding off the rest of the machines until the `WW3_restart_from_binary` thing is worked out.

**Comment by JessicaMeixner-NOAA** [comment]
> WW3_restart_from_binary

done.

File: ush/parsing_ufs_configure.sh
Comment (JessicaMeixner-NOAA) [reply ush/parsing_ufs_configure.sh]: changed

**Comment by WalterKolczynski-NOAA** [comment]
CI Tests set up to run in /lfs/h2/emc/ptmp/walter.kolczynski/PR/PR_3145/RUNTESTS on WCOSS

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**
